webpack/assets/javascripts/react_app/routes/index.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Switch, Route } from 'react-router-dom';
+
+export const pages = []; // React Pages
+
+let currentLocation = null;
+export default (
+  <Switch>
+    {pages.map(({ render, ...props }) => (
+      <Route
+        {...props}
+        key={props.path}
+        render={renderProps => {
+          const railsApplicationContent = document.getElementById(
+            'rails-application-content'
+          );
+          if (railsApplicationContent) {
+            railsApplicationContent.remove();
+          }
+          currentLocation = renderProps.location;
+          return render(renderProps);
+        }}
+      />
+    ))}
+    <Route
+      render={({ location }) => {
+        if (currentLocation && currentLocation.pathname !== location.pathname) {
+          window.Turbolinks.visit(location.pathname);
```

Comment:
from some reason, I"m seeing two turbolink events, when the first one is cancled followed by a real one, it seems its somehow registered multiple times
![image](https://user-images.githubusercontent.com/20167/53662113-c7ee6480-3c6a-11e9-974b-ae1e4db26a65.png)